packages/playwright-core/src/server/recorder.ts

```diff
@@ -243,13 +245,38 @@ export class Recorder implements InstrumentationListener, IRecorder {
     this._refreshOverlay();
   }
 
+  private async _scopeHighlightedSelectorToFrame(frame: Frame): Promise<string> {
+    try {
+      const mainFrame = frame._page.mainFrame();
+      const resolved = await mainFrame.selectors.resolveFrameForSelector(this._highlightedSelector);
+      // selector couldn't be found, don't highlight anything
+      if (!resolved)
+        return '';
+
+      // selector points to no specific frame, highlight in all frames
+      if (resolved?.frame === mainFrame)
+        return stringifySelector(resolved.info.parsed);
```

Comment:
Note: you can add `selector` property to the result of `resolveFrameForSelector()`, it is already available there.

Reply:
I don't see how. `frameChunks` contains parsed selectors, not strings. And we can't use the `selectorString` that's passed into `evaluateHandle`, because it points to frames not to elements in frames. I'll go ahead without this change, let me know how this would work and i'm happy to open a follow-up PR.